src/parser/deserializer.ts

```diff
@@ -334,8 +347,11 @@ function deserializeObject(
         (buffer[index++] << 16) |
         (buffer[index++] << 24);
       const long = new Long(lowBits, highBits);
+      if (useBigInt64) {
+        value = dataview.getBigInt64(0, true);
+      }
       // Promote the long if possible
-      if (promoteLongs && promoteValues === true) {
+      else if (promoteLongs && promoteValues === true) {
```

Comment:
Super nit, it's kinda weird to have the comment break the else if, can you either delete it or move it inside the `promoteLongs && promoteValues === true` condition